specs/protocol/stage-1.md

```diff
@@ -0,0 +1,44 @@
+# Stage 1 Roles and Requirements
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [Permissionless Fault Proofs](#permissionless-fault-proofs)
+- [Roles for Stage 1](#roles-for-stage-1)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+This document outlines configuration requirements (including roles and other parameters)
+for an implementation of the OP Stack to satsify the Stage 1 decentralization requirements as defined
+by
+L2 Beat [[1](https://medium.com/l2beat/introducing-stages-a-framework-to-evaluate-rollups-maturity-d290bb22befe), [2](https://medium.com/l2beat/stages-update-security-council-requirements-4c79cea8ef52)].
+
+## Permissionless Fault Proofs
+
+Stage 1 requires a chain to be operating with Permissionless Fault Proofs.
+
+## Roles for Stage 1
+
+Within the context of an OP Stack, the following roles are required for Stage 1:
+
+1. **Upgrade Controller:** ALthough named for its ability to perform an upgarde, more generally this
+   account is in charge of _safety_, meaning it MUST control any action which has an impact
+   on the determination of a valid L2 state, or the custody of bridged assets.
+
+   This includes upgrading L1 contracts, and modifying the implementation of the dispute game, and
+   any other safety-critical function.
+
+2. **Guardian:** This account is in charge of _liveness_, meaning it MUST control any action which
+   may cause a delay in the finalization of L2 states and the resulting settlement on L1.
```

Comment:
Hmm, we may need to loosen the clear safety vs. liveness phrasing that defines these two roles—the guardian can also change the respected game type, which is more of a safety role than a liveness one. Quoting from the contract:
```solidity
/// @notice Sets the respected game type. Changing this value can alter the security properties of the system,
///         depending on the new game's behavior.
/// @param _gameType The game type to consult for output proposals.
function setRespectedGameType(GameType _gameType) external {
```

Reply:
IMO the phrasing here is right, and if the implementation does't do this then we need to fix it. :)

Fortunately, I did look into this earlier, and determined that although the Guardian can change the game type by calling `setRespectedGameType()`, it is only the Upgrade Controller which can determine the available options by calling `DGF.setImplementation()`. https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/src/dispute/DisputeGameFactory.sol#L199-L201 

I don't think that the natspec is wrong, but it's somewhat incomplete.

Reply:
I guess it's a bit of semantics depending on how you think of it. The UC determines the available options, but the guardian changing from the PDG to FDG does change safety properties. It's just that both were approved by the UC

Reply:
After some offline discussion, I think that there is some room for this semantic discussion, but that the current phrasing reflects our assessment of the stage 1 requirements. 

Are you OK to let it stand as such for now>

Reply:
Actually, I do have an idea for a small rephrasing, standby.

Reply:
You make a good point, however I think it's best to keep this document as is for now.